src/thing_objects.c

```diff
@@ -1694,6 +1694,38 @@ TngUpdateRet object_update_power_lightning(struct Thing *objtng)
 }
 #undef NUM_ANGLES
 
+/**
+ * Finds an empty safe adjacent position on slab.
+ * @param thing The thing which is to be moved.
+  * @param pos The target position pointer.
+ */
+TbBool find_free_position_on_slab(const struct Thing* thing, struct Coord3d* pos)
+{
+    MapSubtlCoord start_stl = ACTION_RANDOM(9);
+    int nav_sizexy = subtile_coord(thing_nav_block_sizexy(thing), 0);
+
+    for (long nround = 0; nround < 9; nround++)
+    {
+        MapSubtlCoord x = start_stl % 3 + thing->mappos.x.stl.num;
+        MapSubtlCoord y = start_stl / 3 + thing->mappos.y.stl.num;
+        if (get_floor_filled_subtiles_at(x, y) == 1)
```

Comment:
what is `==1`?

Reply:
I don't know. It's copied code.

Reply:
It is amount of solid subtiles from bottom. So it should be checked at special locations (garden, portal)

Reply:
I tested blasting specials from water (0) and temple edge (2) and found no issues. Also dropped gold on different heights and noticed no issues there.